### PR TITLE
build(ci): Android specific releases

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-785] 

### Description

Bringing back the diff of https://github.com/artsy/eigen/pull/7913 

Allow releasing specific past versions on Android.

- On uploading a beta to the playstore we will also upload the signed aab to s3
- On promoting a beta to prod we will allow selection/download from s3 and then upload the selected build to the play store

We will need to test if it works as expected to be able to fetch latest 20 android builds from s3.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- ci select 20 Android last releases - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-785]: https://artsyproduct.atlassian.net/browse/PHIRE-785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -209,88 +210,110 @@ lane :ship_beta_android do |options|
...
-      android_artifact_path: "./android/app/build/outputs/bundle/release/app-release.aab"
+      android_artifact_path: './android/app/build/outputs/bundle/release/app-release.aab'
     )
   else
     UI.error("Unknown deployment target: #{deployment_target}")
   end
 end
 
-desc "Promote a play store android beta to production"
+desc 'Promote a play store android beta to production'
```
Comment (gkartalis) [inline fastlane/Fastfile:250]: This is where the changes start, sorry for the verbose diff but added also some linting stuff that my vscode was really persistant in keeping.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -209,88 +210,110 @@ lane :ship_beta_android do |options|
...
+  else
+    UI.user_error!('Stopping the train!')
+  end
+
+  # download the selected build to the typical build output directory
+  output_app_path = '../android/app/build/outputs/bundle/release/app-release.aab'
+  sh('aws s3 cp s3://artsy-citadel/dev/eigen/builds/android/' + selected_build + ' ' + output_app_path)
+
+  # return the selected version num
+  selected_build.scan(/\d+/).first
```
Comment (gkartalis) [inline fastlane/Fastfile:305]: And this is where the changes end. As mentioned above the rest are just linting changes

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (ci select 20 Android last releases - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dedf3863378873b886cb845d192eea46ed5d89d6
</p>

**Review by brainbicycle — CHANGES_REQUESTED** [review]
Thanks for doing, I don't mind most of the linting changes but don't love this pattern of not passing the var explicitly when it is named the same as the arg.


e.g.:
```
  upload_sentry_artifacts(
     sentry_release_name:,
     dist_version: bundle_version,
     platform: 'ios'
   )
```

vs.

```
 upload_sentry_artifacts(
     sentry_release_name: sentry_release_name,
     dist_version: bundle_version,
     platform: "ios"
   )
```

I don't know if it is valid ruby but it is less readable IMO
Can we revert those?

File: fastlane/Fastfile [hunk 3]
```diff
@@ -95,16 +96,16 @@ lane :ship_beta_ios do
   }
 
   pilot(
-    api_key: api_key,
-    beta_app_review_info: beta_app_review_info,
+    api_key:,
+    beta_app_review_info:,
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply fastlane/Fastfile:100]: sounds good to me ;) will do change that back

**Review by brainbicycle — APPROVED** [review]
nice! thanks!

**Pull request merged**